### PR TITLE
add fix-bin helper

### PR DESCRIPTION
**Pull request by mshima**
fixes https://github.com/jhipster/generator-jhipster/issues/25598

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: bin/fix-bin.cjs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env node
+const { chmod } = require('node:fs/promises');
+const { join } = require('node:path');
+const process = require('node:process');
+
+(async () => {
+  if (process.platform !== "win32") {
+    await chmod(join(__dirname, '../dist/cli/jhipster.cjs'), 0o755);
```
Comment (DanielFran) [inline bin/fix-bin.cjs:8]:
```suggestion
    await chmod(join(__dirname, '../dist/cli/jhipster.cjs'), 0755);
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

File: bin/fix-bin.cjs (hunk 1 above)
Comment (mshima) [reply bin/fix-bin.cjs:8]: The octal syntax should work correctly. Is a preference?

Comment (DanielFran) [reply bin/fix-bin.cjs]: Weel seems we cannot use octal
image

File: bin/fix-bin.cjs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env node
+const { chmod } = require('node:fs/promises');
+const { join } = require('node:path');
+const process = require('node:process');
+
+(async () => {
+  if (process.platform !== 'win32') {
+    await chmod(join(__dirname, '../dist/cli/jhipster.cjs'), 0755);
```
Comment (mshima) [inline bin/fix-bin.cjs:8]:
```suggestion
    await chmod(join(__dirname, '../dist/cli/jhipster.cjs'), 0o755);
```

**Review by DanielFran — APPROVED** [review]

**Pull request merged**